Dockerfile

```diff
@@ -3,31 +3,32 @@ FROM rust:latest as builder
 RUN USER=root cargo new --bin lychee
 WORKDIR /lychee
 
-# Just copy the Cargo.toml files and trigger 
+# Just copy the Cargo.toml files and trigger
 # a build so that we compile our dependencies only.
 # This way we avoid layer cache invalidation
 # if our dependencies haven't changed,
 # resulting in faster builds.
 COPY lychee-bin/Cargo.toml lychee-bin/Cargo.toml
 COPY lychee-lib/Cargo.toml lychee-lib/Cargo.toml
-RUN cargo build --release
-RUN rm src/*.rs
+RUN cargo build --release \
+    && rm src/*.rs
```

Comment:
Does the container startup actually abort if one of the `RUN` commands fail? If not, then note that due to `&&` this is not an equivalent change. Use `;` instead to separate the two commands without making the second depend on the success of the first. Same below. `&&` generally only makes sense if you definitely want to rule out the the second command runs without the first returning exit code `0`.

Reply:
Yes this is what we want and yes the docker build command will stop if any RUN fails and you **definitively** want to fail as soon as possible.
Please take a look at the URL that is in my commit (https://github.com/hadolint/hadolint/wiki/DL3059 and https://docs.docker.com/develop/dev-best-practices/).

Reply:
Makes sense then 👍.